src/parser/elements_parser.ts

```diff
@@ -139,15 +150,19 @@ export class ElementsParser {
     obj: NotParsedFabricObject,
     usingElement: Element,
     exactOwner?: Element,
+    processedClipPaths: Set<string> = new Set(),
   ) {
-    const clipPathElements = this.extractPropertyDefinition(
+    const clipPathDefinition = this.extractPropertyDefinition(
       obj,
       'clipPath',
       this.clipPaths,
-    ) as Element[];
-    if (clipPathElements) {
+    );
+    if (clipPathDefinition) {
+      const clipPathElements = clipPathDefinition.def;
       const objTransformInv = invertTransform(obj.calcTransformMatrix());
-      const clipPathTag = clipPathElements[0].parentElement!;
+      const clipPathTag = clipPathElements[0].parentElement!.cloneNode(
+        true,
+      ) as HTMLElement;
```

Comment:
this is new from me, i wanted to create a copy of the clipPath since we are adding to it transforms attributes depending on the element using it